Acrolinx/AcrolinxURLProvider.py

```diff
@@ -37,48 +37,58 @@ class AcrolinxURLProvider(URLGetter):
         "acrolinx_password": {"required": False, "description": "Password for authentication"},
+        "acrolinx_debug": {"required": False, "description": "Prints out debug env info"}
     }
     output_variables = {"url": {"description": "Download URL for Acrolinx."}}
 
     def main(self):
         """Find the download URL"""
+        
         uuid = self.env.get("acrolinx_uuid", None)
         username = self.env.get("acrolinx_username", None)
         password = self.env.get("acrolinx_password", None)
+        
         if uuid == "%acrolinx_uuid%" or uuid == None:
-            uuid = os.environ.get('acrolinx_uuid')
-            if uuid == None:
+            if os.environ.get("acrolinx_uuid") is not None:
+                uuid = os.environ["acrolinx_uuid"]
+            else:
                 raise ProcessorError(
                     "acrolinx_uuid was not provided, fallback to environment variable return None"
                 )
         if username == "%acrolinx_username%" or username == None:
-            username = os.environ.get('acrolinx_username')
-            if username == None:
+            if os.environ.get("acrolinx_username") is not None:
+                username = os.environ["acrolinx_username"]
+            else:
                 raise ProcessorError(
                     "acrolinx_username was not provided, fallback to environment variable return None"
                 )
         if password == "%acrolinx_password%" or password == None:
-            password = os.environ.get('acrolinx_password')
-            if password == None:
+            if os.environ.get("acrolinx_password") is not None:
+                password = os.environ["acrolinx_password"]
+            else:
                 raise ProcessorError(
                     "acrolinx_password was not provided, fallback to environment variable return None"
                 )
-           
         url = URL.format(username, password, uuid)
         cmd = [self.curl_binary(), "--write-out", "'%{json}'", url]
         out, err, code = self.execute_curl(cmd)
         if code != 0:
             raise ProcessorError(
                 f"{cmd} exited non-zero.\n{err}"
             )
-        json_blob = out.split("{")[1]
-        json_blob = "{" + json_blob # add back { so it is valid jsoin
-        json_blob = json_blob.rstrip("'")
-        self.output(json_blob)
-        d = json.loads(json_blob)
-        url = d['redirect_url']
-        self.output(f"Found URL: {url}")
-        self.env["url"] = url
+        try:
+            # regex to match a url
+            regex = r"https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b(?:[-a-zA-Z0-9()@:%_\+.~#?&\/=]*)"
+            match = re.search(regex, out)
+
+            url = match[0]
+            self.output(f"Found URL: {url}")
+            self.env["url"] = url
+        except:
+            raise ProcessorError(
+                "url not found"
```

Comment:
Is there any debug output we can automatically print out here so operators can get an idea of what went wrong?

Reply:
good call. I added better output.